gsy_framework/community_datasheet/community_datasheet_parser.py

```diff
@@ -0,0 +1,182 @@
+
+    @staticmethod
+    def _get_member_coordinates(
+            member_details: Dict, location_converter: LocationConverter, session=None):
+        address = member_details["Location/Address (optional)"] or ""
+        zip_code = member_details["ZIP code"] or ""
+        full_address = f"{address} {zip_code}"
+        if not full_address.strip():
+            return None
+
+        try:
+            return location_converter.convert(full_address, session=session)
+        except LocationConverterException as ex:
+            raise CommunityDatasheetException(ex) from ex
+
+    @staticmethod
+    def _validate_grid(grid):
+        try:
+            scenario_validator(grid)
+        except ValidationError as ex:
+            raise CommunityDatasheetException(ex) from ex
+
+    def _get_assets_by_member(self) -> Dict:
+        """Return a mapping between each member and their assets."""
+        assets_by_member = {member_name: [] for member_name in self._datasheet.members}
+        asset_items = [
+            assets.items()
+            for assets in [self._datasheet.loads, self._datasheet.pvs, self._datasheet.storages]
+        ]
+        for member_name, assets in chain.from_iterable(asset_items):
+            assets_by_member[member_name].extend(assets)
+
+        if any((missing := member) not in self._datasheet.members for member in assets_by_member):
+            raise CommunityDatasheetException(
+                f'Member "{missing}" was defined in one of the asset sheets'  # noqa: F821
+                'but not in the "Community Members" sheet.')
+
+        return assets_by_member
+
+    @staticmethod
+    def _merge_profiles_into_assets(profiles: Dict, assets_by_member: Dict) -> None:
+        """Merge (in-place) each energy profile into the representation of its own asset."""
+        for asset in chain.from_iterable(assets_by_member.values()):
+            asset_name = asset["name"]
+            if asset_name not in profiles:
+                continue
+            asset_type = asset["type"]
+            try:
+                profile_key = PROFILE_KEYS_BY_TYPE[asset_type]
+            except KeyError as ex:
+                raise CommunityDatasheetException(
+                    f'Asset type "{asset_type}" is not supported.') from ex
+            asset[profile_key] = profiles[asset_name]
+
+    def _create_grid(self, assets_by_member: Dict) -> Dict:
+        members_grid = []
+        for member_name, assets in assets_by_member.items():
+            member_representation = self._create_member_representation(member_name)
+            member_representation["children"] = assets
+            members_grid.append(member_representation)
+
+        return {
+            "name": "Grid Market",
+            "tags": None,
+            "type": "Area",
+            "uuid": str(uuid.uuid4()),
+            "children": members_grid
+        }
+
+    def _create_member_representation(self, member_name: str) -> Dict:
```

Comment:
This creates the grid representation of the "member" of the community datasheet. A member is basically a home and can have children (only assets, without further nesting).

Reply:
I have the feeling that we are supposed to use the name that was specified by the user in the datasheet (on the "Community Members" sheet). I'm saying this because the name on the grid will be shown in the UI, and if the names don't match this might be confusing. Do you have some specific reason to suggest using "Home"?

Reply:
Ok, I see, so the nomenclature changed yet again :-( 
It was decided to call the members of the community "Homes" at some point (also reflected in the new UI). 
But apparently, now it is "Member". Could you please double check though whether this change in wording is wanted?
Not really important though, just had to remind myself every time that "member" is the house and not an asset when reviewing.

Reply:
Ah, I do not mean the name of the members that the user inputs. I mean the name that we use internally (and also in code).

Reply:
Ah sorry, I had misunderstood. To be sure: do you mean that I should rename some `member` occurrences to `home` (when they concern the grid representation)? For example, renaming this method to `_create_home_representation()`?

Reply:
Yes, because of the way it is implemented, these members will always be market areas containing assets: homes.
Of course, if you like to leave it as generic as it is, leave it (calling a very late "NIT" :-))